src/software/ai/evaluation/calc_best_shot.cpp

```diff
@@ -1,20 +1,93 @@
 #include "software/ai/evaluation/calc_best_shot.h"
 
-#include "software/ai/evaluation/calc_best_shot_impl.h"
-#include "software/geom/algorithms/acute_angle.h"
-#include "software/geom/algorithms/multiple_segments.h"
-#include "software/geom/algorithms/projection.h"
-
 std::optional<Shot> calcBestShotOnGoal(const Segment &goal_post, const Point &shot_origin,
-                                       const std::vector<Robot> &robot_obstacles)
+                                       const std::vector<Robot> &robot_obstacles,
+                                       TeamType goal, double radius)
 {
-    // Use shot evaluation function to get the best Shot
-    std::vector<Circle> obs;
-    for (Robot robot : robot_obstacles)
+    size_t max_num_obstacles = robot_obstacles.empty() ? 1 : robot_obstacles.size();
```

Comment:
why do we need a min of 1 obstacle? and curious what that obstacle is? 

also nit: ternary operator is banned in our code base

Reply:
reserving size 0 for vectors is illegal. max_num_obstacles is just for reserving space in the vector

Reply:
`reserve` only increases capacity if the value you pass in is larger than its current size

so passing in 0 to reserve must be legal, you probably don't need this line altogether

Reply:
sorry ya ur right reserving -1 is illegal idk what I was doing here lol

Reply:
done